iati/core/tests/test_rulesets.py

```diff
@@ -187,14 +183,38 @@ class RuleSubclassTestBase(object):
     """A base class for Rule subclass tests."""
 
     @pytest.fixture
-    def basic_rule(self, rule_constructor, valid_case):
+    def xpath_base(self):
+        """Return valid xpath_base."""
+        return '//root_element'
+
+    @pytest.fixture
+    def empty_xpath_base(self):
+        """Empty_xpath_base."""
+        return ''
+
+    @pytest.fixture
+    def basic_rule(self, rule_constructor, valid_case, xpath_base):
         """Instantiate a basic Rule subclass."""
-        xpath_base = 'an xpath'
         return rule_constructor(xpath_base, valid_case)
 
     @pytest.fixture
+    def rule(self, rule_constructor, xpath_base, case_for_is_valid_for):
+        """Instantiate a rule with specific cases for checking the `is_valid_for` function."""
+        return rule_constructor(xpath_base, case_for_is_valid_for)
+
+    @pytest.fixture
+    def rule_empty_xpath_base(self, rule_constructor, valid_case, empty_xpath_base):
+        """Instantiate a basic Rule subclass with an empty xpath_base."""
+        return rule_constructor(empty_xpath_base, valid_case)
+
+    @pytest.fixture
+    def rule_empty_xpath_base_and_paths(self, rule_constructor, empty_xpath_base, empty_path_case):
+        """Instantiate a basic Rule subclass with an empty xpath_base and empty path case."""
+        return rule_constructor(empty_xpath_base, empty_path_case)
+
+    @pytest.fixture
     def rule_constructor(self, rule_type):
-        """Return the constructor for the current Rule type."""
+        """Instantiate the current Rule type."""
```

Comment:
There is a fixture in `TestRuleSubclasses` with the same name. May be worth adding a `Note:` to one or both to clarify what the difference is.